packages/adapter-cloudflare/src/worker.js

```diff
@@ -36,26 +37,30 @@ const worker = {
 				}
 			});
 		} else {
-			// prerendered pages and index.html files
-			pathname = pathname.replace(/\/$/, '') || '/';
-
-			let file = pathname.substring(1);
+			// prerendered pages and /static files
 
 			try {
-				file = decodeURIComponent(file);
-			} catch (err) {
-				// ignore
+				pathname = decodeURIComponent(pathname);
+			} catch {
+				// ignore invalid URI
 			}
 
-			if (
-				manifest.assets.has(file) ||
-				manifest.assets.has(file + '/index.html') ||
-				prerendered.has(pathname)
-			) {
+			const stripped_pathname = pathname.replace(/\/$/, '') || '';
+
+			let is_asset = false;
+			const filename = stripped_pathname.substring(1);
+			if (filename) {
+				is_asset = manifest.assets.has(filename) || manifest.assets.has(filename + '/index.html');
+			}
+
+			const is_prerendered = prerendered.has(pathname) || prerendered.has(stripped_pathname);
```

Comment:
is this check too permissive? if a trailing slash is present when it shouldn't be, or vice versa, we probably want to respond with a redirect rather than `env.ASSETS.fetch`

Reply:
You're right! I hadn't even thought of that. I've now added a redirect if there a trailing slash or non-trailing slash counterpart is found.